src/Uno.UI/UI/ViewManagement/InputPane/InputPaneVisibilityEventArgs.cs

```diff
@@ -1,14 +1,27 @@
+using System;
+using Uno.UI.Controls;
+
 namespace Windows.UI.ViewManagement
 {
 	public partial class InputPaneVisibilityEventArgs
 	{
 		internal InputPaneVisibilityEventArgs(Foundation.Rect occludedRect)
 		{
 			OccludedRect = occludedRect;
+#if __ANDROID__
+			var navigationBarHeight = NavigationBarHelper.IsNavigationBarVisible && !NavigationBarHelper.IsNavigationBarTranslucent
+				? NavigationBarHelper.LogicalNavigationBarHeight
+				: 0;
+			KeyboardHeight = Math.Max(0, OccludedRect.Height - navigationBarHeight);
+#endif
 		}
 
 		public bool EnsuredFocusedElementInView { get; set; }
 
 		public Foundation.Rect OccludedRect { get; private set; }
+
+#if __ANDROID__
```

Comment:
Do we need to add KeyboardHeight on Android? It's not in UWP's contract.

Reply:
Yes because we need it for the InputPaneExtensions

Reply:
Or should I modify only the InputPaneExtensions only (using OccludedRect minus NavigationBar.Height) ?

Reply:
Yes :)